engine/src/main/java/io/zeebe/engine/processing/bpmn/container/CallActivityProcessor.java

```diff
@@ -136,7 +136,12 @@ public void onChildCompleted(
     switch (currentState) {
       case ELEMENT_ACTIVATED:
         stateTransitionBehavior.transitionToCompleting(callActivityContext);
-        stateBehavior.propagateTemporaryVariables(childContext, callActivityContext);
+
+        if (element.getOutputMappings().isPresent()
+            || element.isPropagateAllChildVariablesEnabled()) {
```

Comment:
I'm not sure this if statement is correct. If there are output mappings, but the new property is disabled, then this would still propagate all temp vars.

Reply:
Yes, it is correct :sweat_smile:  We need to set the temp variables to apply the output mappings on completing the call activity. Since this is done in another step (i.e. when processing another record), we need to copy all variables to temp.